ssl/ssl_conf.c

```diff
@@ -671,23 +671,21 @@ static int cmd_RecordPadding(SSL_CONF_CTX *cctx, const char *value)
             OPENSSL_free(copy);
             return 0;
         }
-        hs_padding = (size_t) atoi(commap + 1);
+        hs_padding = (size_t) strtoul(commap + 1, NULL, 0);
+        if ((hs_padding == ULONG_MAX) && (errno == ERANGE))
```

Comment:
This is still not robust.

`strtoul()` -- returns `0 (zero)` if no valid conversion could be performed. Which doesn't help distinguish with the real zero. Also, multiple other error scenarios are not getting handled.

We can certainly make it robust by having our own new function, which could look like something below:

```
    char *endptr;  //  Store the location where conversion stopped
    errno = 0; // this is crucial.
    unsigned long y = strtoul(s.c_str(), &endptr, base);
    
    if (s == endptr) puts("No conversion");      // "", "abc"
    else if (errno == ERANGE) puts("Overflow");
    else if (*endptr) puts("Extra text after the number"); // "0xyz", "0 ", "0.0"
    else if (errno != 0 ) puts("some error happened");
    else puts("100% successful");
```

Reply:
I think (at least for the purposes of this conversion, we could collapse that down to
```
char *endptrl
unsigned long val = strtoul(in, &endptr, 0);
if (s == endptr || errno != 0)
   return 0;
```
Since we don't care about the specific error, only weather or not we got a valid conversion.

I'll fix it up in the am

Reply:
Essentially in short it would look like below:

```
// return 1 on sucess and 0 on failure.
int convertTextToNum(const char * s, unsigned long& y)
{
    char *endptr;  //  Store the location where conversion stopped
    errno = 0; // this is crucial.
    unsigned long y = strtoul(s, &endptr, 0);
    if ( s == endptr || errno != 0 || *endptr )
        return 0;
    return 1;
}
```

I provided detailed code with comments above to aid in understanding.

Reply:
Checking whether *endptr == '\0' or not depends on whether the whole `s` must represent a number or if we are OK with just a prefix being a number. That depends on context. But yeah, we should probably have an utility  function such as the one you wrote above somewhere to be used throughout the code.